File: dotnet/Vaas/src/Vaas/Options/ForStreamOptions.cs
namespace Vaas.Options;

public class ForStreamOptions
{
    public bool UseHashLookup { get; init; } = true;
    public string? VaasRequestId { get; init; }

    public static ForStreamOptions From(VaasOptions options)
    {
        return new ForStreamOptions { UseHashLookup = options.UseHashLookup, VaasRequestId = null };
    }
}

File: typescript/src/options/ForStreamOptions.ts
import { VaasOptions } from "./VaasOptions";

export class ForStreamOptions {
  public useHashLookup: boolean;
  public vaasRequestId?: string;

  constructor(options?: { useHashLookup?: boolean; vaasRequestId?: string }) {
    this.useHashLookup = options?.useHashLookup ?? true;
    this.vaasRequestId = options?.vaasRequestId;
  }

  public static from(options: VaasOptions): ForStreamOptions {
    return new ForStreamOptions({
      useHashLookup: options.useHashLookup,
    });
  }
}
